granian/server.py

```diff
@@ -461,9 +465,50 @@ def setup_signals(self):
         if sys.platform != 'win32':
             signal.signal(signal.SIGHUP, self.signal_handler_reload)
 
-    def startup(self, spawn_target, target_loader):
-        logger.info(f'Starting granian (main PID: {os.getpid()})')
+    def _write_pid(self):
+        with self.pid.open('w+') as pid_file:
+            pid_file.write(f'{self.main_pid}\n')
+
+    def _write_pidfile(self):
+        if not self.pid:
+            return
+
+        if self.pid.exists():
+            with self.pid.open('r') as pid_file:
+                try:
+                    old_pid = int(pid_file.read())
+                except ValueError:
+                    self._write_pid()
+                    return
+            try:
+                os.kill(old_pid, 0)
+            except OSError as e:
+                if e.args[0] == errno.ESRCH:
+                    self._write_pid()
+                    return
+                else:
+                    raise e
```

Comment:
This looks a bit too overcomplicated to me..
I'd say Granian shouldn't try to kill other processes; checking if the file exists before writing and raising an error in such case is more than enough IMHO.

Reply:
`os.kill(old_pid, 0)` is not trying to kill the particular process. Its just trying to find if the particular pid exists in system, If not we just overwrite the `pid_file` with the new pid .

But If you don't want to handle this case. I'll be happy to remove that change from this PR..

Reply:
Understood.
Let's keep it, but change line 481 https://github.com/emmett-framework/granian/pull/327/files/405ff32e5fbe32a1ac6db956e6d9df2c71913be8#diff-571aaa34f48968993dca3f59fcc89aff6e892451c088006d134ab11c0acf0cc2R481 to be more conservative: if we cannot read the pid from that file, let's just stop and raise, 'cause probably it's not written by Granian.

Reply:
Addressed via https://github.com/emmett-framework/granian/compare/405ff32e5fbe32a1ac6db956e6d9df2c71913be8..ea4efa7dad1e4b1fbe09d2dfd822936be9d2095d